dashboard/app/controllers/followers_controller.rb

```diff
@@ -52,42 +24,32 @@ def remove
 
     authorize! :destroy, f
     f.delete
-    FollowerMailer.student_disassociated_notify_teacher(@teacher, @user).deliver_now if @teacher.email.present?
-    redirect_to root_path, notice: t('teacher.student_teacher_disassociated', teacher_name: @teacher.name, student_name: @user.name)
+    FollowerMailer.student_disassociated_notify_teacher(@teacher, current_user).deliver_now if @teacher.email.present?
+    redirect_to root_path, notice: t('teacher.student_teacher_disassociated', teacher_name: @teacher.name, student_name: current_user.name)
   end
 
   # GET /join/XXXXXX
-  # join a section as a new student
+  # if logged in, join the section, if not logged in, present a form to create a new user and log in
   def student_user_new
-    @section = Section.find_by_code(params[:section_code])
+    load_section
 
     # make sure section_code is in the path (rather than just query string)
     if request.path != student_user_new_path(section_code: params[:section_code])
       redirect_to student_user_new_path(section_code: params[:section_code])
     elsif current_user && @section
-      if current_user == @section.user
-        redirect_to root_path, alert: I18n.t('follower.error.cant_join_own_section') and return
```

Comment:
This works (with load_section as the before_filter) because before filters return if performed already: http://guides.rubyonrails.org/action_controller_overview.html#filters